test/unit/c_api/eigensolver/test_eigensolver_c_api.cpp

```diff
@@ -135,7 +136,7 @@ void testEigensolver(const blas::Uplo uplo, const SizeType m, const SizeType mb,
       else {
         DLAF_ASSERT(false, typeid(T).name());
       }
-      EXPECT_EQ(0, err);
+      DLAF_ASSERT(err == 0, err);
```

Comment:
Is this change simply meant to abort if `err` isn't zero, or is there another reason for changing this from `EXPECT_EQ` to `DLAF_ASSERT`?

Reply:
Yes, I think it's safer to abort. I think it should have been `ASSERT_EQ`/DLAF_ASSERT` from the beginning.

Reply:
Not sure which is the best candidate here, but remind that:
- `DLAF_ASSERT` stops the entire test executable
- `ASSERT_EQ` stops the current gtest case
- `EXPECT_EQ` marks the test as failed and continues